src/librbd/crypto/CryptoImageDispatch.h

```diff
@@ -0,0 +1,117 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
```

Comment:
Nit: I would add a CC file so that you can explicitly instantiate it and verify it compiles.